lightkurve/utils.py

```diff
@@ -484,22 +484,27 @@ def detect_filetype(header):
     try:
         # use `telescop` keyword to determine mission
         # and `creator` to determine tpf or lc
-        telescop = header['telescop'].lower()
+        # Some old custom TESS data did not define the `TELESCOP` card
+        if 'TELESCOP' in header.keys():
+            telescop = header['telescop'].lower()
+        else:
+            telescop = header['mission'].lower()
         creator = header['creator'].lower()
         origin = header['origin'].lower()
         if telescop == 'kepler':
             # Kepler TPFs will contain "TargetPixelExporterPipelineModule"
             if 'targetpixel' in creator:
                 return 'KeplerTargetPixelFile'
             # Kepler LCFs will contain "FluxExporter2PipelineModule"
-            elif 'fluxexporter' in creator or 'lightcurve' in creator:
+            elif ('fluxexporter' in creator or 'lightcurve' in creator
+                or 'to_fits()' in creator):
                 return 'KeplerLightCurveFile'
         elif telescop == 'tess':
             # TESS TPFs will contain "TargetPixelExporterPipelineModule"
             if 'targetpixel' in creator:
                 return 'TessTargetPixelFile'
             # TESS LCFs will contain "LightCurveExporterPipelineModule"
-            elif 'lightcurve' in creator:
+            elif 'lightcurve' in creator or 'to_fits()' in creator:
```

Comment:
Can we simplify to:
```elif 'lightcurve' in creator.lower()```

Reply:
I'm worried that some day we'll have a TPF.to_fits(), we can prolly just remove to_fits here and it;ll still work.